src/ui/public/vislib/lib/axis_config.js

```diff
@@ -0,0 +1,108 @@
+import _ from 'lodash';
+export default function AxisConfigFactory() {
+
+  const defaults = {
+    show: true,
+    type: 'value',
+    elSelector: '.axis-wrapper-{pos} .axis-div',
+    position: 'left',
+    scale: {
+      type: 'linear',
+      expandLastBucket: true, //TODO: rename ... bucket has nothing to do with vis
+      inverted: false
+    },
+    style: {
+      color: '#ddd',
+      lineWidth: '1px',
+      opacity: 1,
+      tickColor: '#ddd',
+      tickWidth: '1px',
+      tickLength: '6px'
+    },
+    labels: {
+      axisFormatter: null, // TODO: create default axis formatter
+      show: true,
+      rotate: 0,
+      rotateAnchor: 'center',
+      filter: false,
+      color: '#ddd',
+      font: '"Open Sans", "Lato", "Helvetica Neue", Helvetica, Arial, sans-serif', // TODO
+      fontSize: '8pt',
+      truncate: 100
+    },
+    title: {
+      text: '',
+      elSelector: '.axis-wrapper-{pos} .axis-title'
+    }
+  };
+
+  const categoryDefaults = {
+    type: 'category',
+    position: 'bottom',
+    labels: {
+      rotate: 0,
+      rotateAnchor: 'end',
+      filter: true
+    }
+  };
+
+  class AxisConfig {
+    constructor(config) {
+      if (config.type === 'category') {
+        _.defaultsDeep(config, categoryDefaults);
```

Comment:
This is modifying the `config` variable, which probably isn't a good idea because the `AxisConfig` doesn't know where it came from or what it's purpose it.

We should probably start off the constructor with `this._values = {}` and then apply the defaults:

```js
const typeDefaults = config.type === 'category' ? categoryDefaults : {}
this._values = _.defaultsDeep({}, config, typeDefaults, defaults);
```